src/makeInfinityAdder.js

```diff
@@ -4,7 +4,25 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let flag = 0;
+  let sum = 0;
+
+  const makeSum = (a) => {
+    if (typeof a === 'number') {
+      sum += a;
+      flag = 0;
+
+      return makeSum;
+    } else if (typeof a !== 'number') {
```

Comment:
The check for 'typeof a !== 'number'' is unnecessary because the else block will naturally handle any case where 'a' is not a number. You can simplify the code by removing this condition.